src/preferences/dialog/dlgprefrecord.cpp

```diff
@@ -135,125 +109,81 @@ DlgPrefRecord::DlgPrefRecord(QWidget* parent, UserSettingsPointer pConfig)
+    connect(SliderQuality, SIGNAL(sliderMoved(int)),
+            this, SLOT(slotSliderQuality()));
+    connect(SliderQuality, SIGNAL(sliderReleased()),
+            this, SLOT(slotSliderQuality()));
+    connect(SliderCompression, SIGNAL(valueChanged(int)),
+            this, SLOT(slotSliderCompression()));
+    connect(SliderCompression, SIGNAL(sliderMoved(int)),
+            this, SLOT(slotSliderCompression()));
+    connect(SliderCompression, SIGNAL(sliderReleased()),
+            this, SLOT(slotSliderCompression()));
 }
 
-void DlgPrefRecord::slotEncoding() {
-    // set defaults
-    groupBoxQuality->setEnabled(true);
-    //m_pConfig->set(ConfigKey(RECORDING_PREF_KEY, "Encoding"), ConfigValue(comboBoxEncoding->currentText()));
-
-    if (m_pRadioWav && m_pRadioWav->isChecked()) {
-        m_pConfig->set(ConfigKey(RECORDING_PREF_KEY, "Encoding"), ConfigValue(ENCODING_WAVE));
-        groupBoxQuality->setEnabled(false);
-    } else if (m_pRadioFlac && m_pRadioFlac->isChecked()) {
-        m_pConfig->set(ConfigKey(RECORDING_PREF_KEY, "Encoding"), ConfigValue(ENCODING_FLAC));
-        groupBoxQuality->setEnabled(false);
-    } else if (m_pRadioAiff && m_pRadioAiff->isChecked()) {
-        m_pConfig->set(ConfigKey(RECORDING_PREF_KEY, "Encoding"), ConfigValue(ENCODING_AIFF));
-        groupBoxQuality->setEnabled(false);
-    } else if (m_pRadioOgg && m_pRadioOgg->isChecked()) {
-        int value = m_pConfig->getValueString(ConfigKey(RECORDING_PREF_KEY, "OGG_Quality")).toInt();
-        // If value == 0 then a default value of 128kbps is proposed.
-        if (!value)
-            value = 6; // 128kbps
-
-        SliderQuality->setValue(value);
-        m_pConfig->set(ConfigKey(RECORDING_PREF_KEY, "Encoding"), ConfigValue(ENCODING_OGG));
-    } else if (m_pRadioMp3 && m_pRadioMp3->isChecked()) {
-        int value = m_pConfig->getValueString(ConfigKey(RECORDING_PREF_KEY, "MP3_Quality")).toInt();
-        // If value == 0 then a default value of 128kbps is proposed.
-        if (!value) {
-            value = 6;  // 128kbps
+DlgPrefRecord::~DlgPrefRecord() {
+    // Note: I don't disconnect signals, since that's supposedly done automatically
+    // when the object is deleted
+    foreach(QRadioButton* button, m_formatButtons) {
+        if (LosslessEncLayout->indexOf(button) != -1) {
+            LosslessEncLayout->removeWidget(button);
+        } else {
+            LossyEncLayout->removeWidget(button);
         }
-
-        SliderQuality->setValue(value);
-        m_pConfig->set(ConfigKey(RECORDING_PREF_KEY, "Encoding"), ConfigValue(ENCODING_MP3));
-    } else {
-        qDebug() << "Invalid recording encoding type in" << __FILE__ << "on line:" << __LINE__;
+        // TODO: Not sure if this is necessary or correct, or I should simply "delete button;"
+        emit(button->deleteLater());
     }
+    foreach(QAbstractButton* widget, m_optionWidgets) {
+        OptionGroupsLayout->removeWidget(widget);
+        emit(widget->deleteLater());
```

Comment:
You can probably get around of all this delete and reconnect calls, if you just create all buttons parented to the containing widget. and just set them visible as required. The qt object tree will than care for the delete calls.